x/gamm/keeper/pool.go

```diff
@@ -79,6 +79,80 @@ func (k Keeper) SetPool(ctx sdk.Context, pool types.PoolI) error {
 	return nil
 }
 
+// CleanupBalancerPool destructs a pool and refund all the assets according to
+// the shares held by the accounts. CleanupBalancerPool should be called not during
+// the chain execution time, as it iterates the entire account balances.
+//
+// All locks on this pool share must be unlocked in prior. Execute LockupKeeper.ForceUnlock
+// on remaning locks before calling this function.
+func (k Keeper) CleanupBalancerPool(ctx sdk.Context, poolId uint64) (err error) {
+	pool, err := k.GetPool(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
+	totalShares := pool.GetTotalShares().Amount
+	shareDenom := pool.GetTotalShares().Denom
+	poolAssets := pool.GetAllPoolAssets()
+
+	// first iterate through the share holders and burn them
+	k.bankKeeper.IterateAllBalances(ctx, func(addr sdk.AccAddress, coin sdk.Coin) (stop bool) {
```

Comment:
Lets file an issue to revisit performance of this. (Either we change to do one iteration for all pools, or we wait for https://github.com/cosmos/cosmos-sdk/pull/9611 to get merged into a release)

Reply:
Wait aren't we going to be iterating over IBC's module accounts like this? If so, we will be making IBC'd tokens never redeemable. Do we need to care about this, cc @sunnya97 ?

Reply:
In the original issue I think the IBC'd LP tokens are never redeemable for now(also there is no use case of IBCing out LP tokens yet). I could apply the first and the third suggestion now.